### ISSUE TITLE
[AVM Module Issue]: Unable to set non-compliance message using policy_assignments_to_modify

### ISSUE THREAD

**Issue by JWilkinsonMB** [Azure/terraform-azurerm-avm-ptn-alz#87]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### (Optional) Module Version

v0.8.0

### Description

Unable to set the non-compliance message using the policy_assignments_to_modify variable. 
If set it has no effect and the default value from the policy assignment template is used instead. 

E.g.:
```hcl
policy_assignments_to_modify = {
    global = {
      policy_assignments = {
        Deny-Resource-Locations = {
          parameters = {
            listOfAllowedLocations = jsonencode({ value = local.allowed_locations })
          }
          non_compliance_message = [{
            message = "The resource location is not allowed. Please select a different location. Allowed locations are: ${join(", ", local.allowed_locations)}"
          }]

        }
      }
    }
  }
```

The reason is that this module defines `non_compliance_message`, however the ALZ provider is expecting `non_compliance_messages` (plural).

Updating the variable to use non_compliance_message**s** resolves the issue (docs would also need updating):

https://github.com/Azure/terraform-azurerm-avm-ptn-alz/blob/fdf3f8767d9182fa604d11ef9bb08d13ece857dd/variables.tf#L54-L57

https://github.com/Azure/terraform-provider-alz/blob/main/docs/data-sources/architecture.md#nested-schema-for-policy_assignments_to_modifypolicy_assignments

**Comment by matt-FFFFFF** [comment]
Thank you for raising this issue. We will fix it!

**Comment by matt-FFFFFF** [comment]
fixed in v0.8.1
